tests/strict_notnull_tests.cpp

```diff
@@ -43,12 +52,116 @@ GSL_SUPPRESS(f.4) // NO-FORMAT: attribute
 // clang-format on
 bool strict_helper_const(strict_not_null<const int*> p) { return *p == 12; }
 
-#ifdef CONFIRM_COMPILATION_ERRORS
 int* return_pointer() { return nullptr; }
+#ifdef CONFIRM_COMPILATION_ERRORS
```

Comment:
This is unfamiliar and I can't figure out who/what sets it. Can you explain `CONFIRM_COMPILATION_ERRORS` to me?

Reply:
[nevermind](https://github.com/microsoft/GSL/issues/1130#issuecomment-1699857287)

Reply:
If you want, I can write test that check for non-compiling constructs like the ones written in the enable_if-PR. That could be done step by step (whenever I have time) for all the stuff that is right now hidden in those `#ifdef CONFIRM_COMPILATION_ERRORS`.

Reply:
It would be good to have regression tests that make sure we reject invalid uses where possible.

Reply:
First shot: https://github.com/microsoft/GSL/pull/1180